lib/taskana-core/src/test/java/pro/taskana/ArchitectureTest.java

```diff
@@ -84,20 +83,92 @@ void noClassShouldAccessStandardStreams() {
     NO_CLASSES_SHOULD_ACCESS_STANDARD_STREAMS.check(importedClasses);
   }
 
-  @Test
+  /**
+   * Solution would be nice, but is not achievable without big changes
+   * https://www.archunit.org/userguide/html/000_Index.html#_cycle_checks
+   */
   @Disabled
+  @Test
   void freeOfCycles() {
-    ArchRule myRule = slices().matching("pro.taskana.(*)..")
-                          .should().beFreeOfCycles();
+    ArchRule myRule = slices().matching("pro.taskana.(*)..").should().beFreeOfCycles();
     myRule.check(importedClasses);
   }
 
+  /**
+   * Test for cycles with subpackages
+   * https://www.archunit.org/userguide/html/000_Index.html#_cycle_checks
+   */
+  @TestFactory
+  Collection<DynamicTest> freeOfCycles_subpackages() {
+
+    Stream<String> packagesToTest =
+        Stream.of(
+            "pro.taskana.common.internal.(*)..",
+            "pro.taskana.common.api.(*)..",
+            "pro.taskana.common.internal.(*)..",
```

Comment:
Why "pro.taskana.common.internal.(*).." twice?

Reply:
safe is safe ... 
jk. done.